docs/plugins/development.md

```diff
@@ -0,0 +1,243 @@
+# Plugin Development
+
+This documentation covers the development of custom plugins for NetBox. Plugins are essentially self-contained [Django apps](https://docs.djangoproject.com/en/stable/) which integrate with NetBox to provide custom functionality. Since the development of Django apps is already very well-documented, we'll only be covering the aspects that are specific to NetBox.
+
+## Initial Setup
+
+## Plugin Structure
+
+Although the specific structure of a plugin is largely left to the discretion of its authors, a typical NetBox plugin looks something like this:
+
+```no-highlight
+plugin_name/
+  - plugin_name/
+    - templates/
+      - *.html
```

Comment:
I think we should recommend the practice of nesting templates within an app directory as that seems to be best practice and helps prevent namespace collisions when django searches for templates:
```
templates/
  plugin_name/
    *.html
```

Reply:
IMO that seems redundant. Can you give an example of a problem that would prevent?

Reply:
If I develop a plugin that injects content into say the device and IP address pages, it is reasonable that I might do this:
```
templates/
    dcim/
        device.html
    ipam
        ipaddress.html
```
When django resolves the render command with the string `"dcim/device.html"` this created a namespace collision within core and in fact, in this case, django tries to render our core template a second time but fails due to lack of context.

I can't find any specific django docs but it does seem to be a general convention with a purpose. All of these apps to it: [oauth2_provider](https://github.com/jazzband/django-oauth-toolkit/tree/master/oauth2_provider), [debug_toolbar](https://github.com/jazzband/django-debug-toolbar/tree/master/debug_toolbar), [pipeline](https://github.com/jazzband/django-pipeline/tree/master/pipeline)

Reply:
Oh, I see. I was thinking between core and plugin, rather than between two plugins.